demos/metrics_demo.cc

```diff
@@ -0,0 +1,137 @@
+#include <seastar/util/defer.hh>
+#include <seastar/core/metrics.hh>
+#include <seastar/core/metrics_api.hh>
+#include <seastar/core/metrics_registration.hh>
+#include <fmt/format.h>
+
+using namespace seastar;
+using namespace std::chrono_literals;
+namespace sm = seastar::metrics;
+namespace bpo = boost::program_options;
+
+logger applog("app");
+
+class metrics_example {
+private:
+    sm::metric_groups _metrics;
+    uint64_t _counter;
+    uint64_t _triggered;
+
+public:
+    metrics_example(int instance) : _counter(0) {
+        // Note that these metrics are only available as long at the _metrics instance
+        // is valid.
+        _metrics.add_group(fmt::format("metrics_demo_{}", instance), {
+            // Access the member each time the metric is read.
+            // Note that counter is expected to be monotonically increasing.
+            sm::make_counter("counter", _counter, sm::description("A counter that counts.")),
+            // A function that can return any integer
+            sm::make_counter("function", [this]() {applog.info("triggered"); return ++_triggered;}, sm::description("A function which returns a value.")),
+            // A gauge is a counter that can go up and down, can be used for rates
+            sm::make_gauge("gauge", []() {return 3;}, sm::description("random number"))
+        });
+    }
+
+    void increase(uint64_t val = 1) { _counter += val; }
+};
+
+class prometheus_example {
+private:
+    httpd::http_server_control _server;
+    bool _was_started;
+public:
+    prometheus_example(sstring address, uint16_t port, sstring prefix) {
+        prometheus::config pctx;
+        net::inet_address prom_addr(address);
+        pctx.metric_help = "seastar::httpd server statistics";
+
+        std::cout << "starting prometheus API server" << std::endl;
+        _server.start("prometheus").get();
+        prometheus::start(_server, pctx).get();
+        _was_started = true;
+
+        _server.listen(socket_address{prom_addr, port}).handle_exception([prom_addr, port] (auto ep) {
+            std::cerr << seastar::format("Could not start Prometheus API server on {}:{}: {}\n", prom_addr, port, ep);
+            return make_exception_future<>(ep);
+        }).get();
+    }
+
+    ~prometheus_example() {
+        if (_was_started) {
+            std::cout << "Stoppping Prometheus server" << std::endl;  // This can throw, but won't.
+            _server.stop().get();
+        }
+    }
+};
+
+/**
+ * In order to access the metrics we need to use one of the builtin mechanisms.
+ * For the sake of this example we are using prometheus.
+ * To view the specific metrics of the demo use something in the lines of
+ *  curl http://localhost:9180/metrics | grep "metrics_demo"
+ * 
+ */
+int main(int argc, char** argv) {
+    seastar::app_template app;
+    app.add_options()("prometheus_port", bpo::value<uint16_t>()->default_value(9180), "Prometheus port. Set to zero in order to disable.");
```

Comment:
You can use metrics relabel to turn off all metrics besides the one used for Demo, not sure if it makes things clearer or not.

Reply:
Could you give an example for how the relabel process?

From the config docs, it seems like its not intended to be used, so I guess shouldn't be included in the example usage?
```cpp
/*!
 * Holds prometheus related configuration
 */
struct config {
    sstring metric_help; //!< Default help message for the returned metrics
    sstring hostname; //!< hostname is deprecated, use label instead
    std::optional<metrics::label_instance> label; //!< A label that will be added to all metrics, we advice not to use it and set it on the prometheus server
    sstring prefix = "seastar"; //!< a prefix that will be added to metric names
};
```